main.js

```diff
@@ -0,0 +1,220 @@
+    baseHeader.textContent = 'Base Name';
+    nameHeader.textContent = 'Rates Name';
+    headerRow.appendChild(baseHeader);
+    headerRow.appendChild(nameHeader);
+    const rateHeader = document.createElement('th');
+    rateHeader.textContent = 'Rates';
+    headerRow.appendChild(rateHeader);
+    tbody.appendChild(headerRow);
+  
+    //Create rows for each currency and its rates
+    currencyData.forEach(currency => {
+      for (const currencyCode in currency.rates) {
+        if (currency.rates.hasOwnProperty(currencyCode)) {
+          const row = document.createElement('tr');
+          const baseCell = document.createElement('td');
+          baseCell.textContent = currency.baseCurrency;
+          row.appendChild(baseCell);
+          const nameCell = document.createElement('td');
+          nameCell.textContent = currencyCode;
+          row.appendChild(nameCell);
+          const rateCell = document.createElement('td');
+          rateCell.textContent = currency.rates[currencyCode];
+          row.appendChild(rateCell);
+          tbody.appendChild(row);
+        }
+      }
+    });
+  
+    //Append the table to the container
+    table.appendChild(tbody);
+    tableContainer.innerHTML = '';
+    tableContainer.appendChild(table);
+  }
+  
+
+//Function to convert currency
+function convertCurrency() {
+  let mainCurrency = document.getElementById('mainCurrency').value;
+  let targetCurrency = document.getElementById('targetCurrency').value;
+  let amount = parseFloat(document.getElementById('amount').value);
+
+  //Validate the amount
+  if (isNaN(amount)){
+    document.getElementById('result').innerText = 'Please enter a valid amount';
+    return;
+  }
+
+  //Find the index of the main currency in the currencyData array
+  const mainCurrencyIndex = currencyData.findIndex(currency => currency.baseCurrency === mainCurrency);
+
+  if (mainCurrencyIndex !== -1) {
+    //Check if the target currency is in the rates for the main currency
+    if (currencyData[mainCurrencyIndex].rates.hasOwnProperty(targetCurrency)) {
+      const conversionRate = currencyData[mainCurrencyIndex].rates[targetCurrency];
+      const result = amount * conversionRate;
+
+      document.getElementById('result').innerText = `${amount} ${mainCurrency} equals ${result.toFixed(2)} ${targetCurrency}`;
+
+      return result;
+    } else {
+      document.getElementById('result').innerText = 'Conversion rate not available for the target currency';
+    }
+  } else {
+    document.getElementById('result').innerText = 'Main currency not found';
+  }
+}
+
+//Search Function Week2
+function searchCurrency() {
+    const searchFrom = document.getElementById('searchFrom').value;
+    const searchResultContainer = document.getElementById('searchResult');
+
+    //console.log('Search from:', searchFrom);
+
+    //Find the base currency in the currencyData array
+    const baseCurrency = currencyData.find(currency => currency.baseCurrency === searchFrom);
```

Comment:
The search form should serve to filter currency based on rates. eg: you have a table 
**-------------------
CZK|USD|0.2
DKK|CZK|0.5
USD|EUR|1.0
-------------------**
And you want to print out results with rates between　(from)0.4 and (to)2.0 so the result would be 
**-------------------
DKK|CZK|0.5
USD|EUR|1.0
-------------------**